### DISCUSSION TITLE
Custom LUTs for fixing lack of black in OBS source capture

### DISCUSSION THREAD

**Discussion by NOVAinc** [pubpub-zz/ppInk#128] [Ideas]
@pubpub-zz I set up to make the LUT. I made 2 versions. Version 1 darkens 90% gray to be black. It shouldn't affect anything else so no color changes, just 90% dark grays. The upside of this is you can display almost black if you apply the lut after you color key the black background out. The downside is that because of the shark color keying, it's not possible to smooth out the keying and as a result there is a black halo effect on stroke edges because of antialiasing.

I have made a second LUT which makes 50%< blacks almost pure black as well, and then extends 0 to 49% black to cover the full gray spectrum. This means a 25% black using the second LUT becomes 50% black, and a 45% black becomes 90%. The advantage of this is that you can include soft blacks (like drop shadows) in your cliparts using 50 to 99% black and smooth the color keying so it allows for the shadow effect.

None of these LUTs will fix clipart transparency when capturing ppink as an independent source within OBS, but they both serve a specific purpose each. Also none of these LUTs messes with any colors, just levels of black.

I have created a test picture with some cliparts and strokes. The top cubes have drop shadows using 50/60/70/80/90/100% black respectively. Then I added strokes on many basic colors, as well as 50% black, 90% black and 25% black (last 3 strokes) This is OBS capturing ppink as a source. As expected, the background appears black.
![standard](https://user-images.githubusercontent.com/19545604/126002662-8b6bfef2-9305-4eea-8e29-0984d68ef57b.png)

This is the same source with a color key applied to it. The color key simply removes the background black. The settings for it are:
Key color: #000000
Similarity: 1
Smoothness: 1

After the color key, LUT (attached below) is applied and darkens the 90% black stroke (98% black with LUT), while it leaves 25% black untouched.
![standard+lut1](https://user-images.githubusercontent.com/19545604/126004472-666164f9-3d10-499a-8472-3038c3fcf01c.png)

This is the same source with a color key with different settings. This time around the color key removes black and smooths out 50%+ black levels so things like smooth shadow look good. The settings for it are:
Key color: #000000
Similarity: 1
Smoothness: 500

After the color key, smooth LUT (attached below) is applied to it to smooth out the shadows and remove some of the halo effect around strokes. This also makes the 90% black stroke a bit translucent (second to last, looks lighter because of the white background). In addition, this one extends 0-49% black to cover almost the full spectrum, so the 50% and 25% black strokes went up to around 100% and 50% black respectively (the third to last and last strokes).
![standard+lut2](https://user-images.githubusercontent.com/19545604/126004790-842b8ca7-f412-415f-a788-538fa5fd6060.png)

Here are the LUTs
LUT
![LUT](https://user-images.githubusercontent.com/19545604/126004832-7f14f3d2-d1f8-4228-b0ac-f7f89de3ffe5.png)

Smooth LUT
![smooth LUT](https://user-images.githubusercontent.com/19545604/126004840-6f7b7726-6a87-4474-8054-cd3510d9081a.png)

I tried my best to make these as precise as I could. Feel free to do with these anything you want, including packaging it with the app if you want. If you have any requests let me know!

**Comment by pubpub-zz** [comment]
This looks great! I will test it this weekend
I will certainly insert it and convert this post into a discussion to make it last.
